### PR TITLE
Add Adam and AdamW optimizers to torch_nntile

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Adds fused Adam and AdamW optimizer support to `torch_nntile`, following the same pattern as the existing SGD integration.

## Changes

- **C++ bindings**: `_C.adam_step` and `_C.adamw_step` call libnntile `tensor::adam_step` / `tensor::adamw_step` via new executor helpers (`tensor_adam_step_fp32`, `tensor_adamw_step_fp32`).
- **Python API** (`torch_nntile.training`):
  - `Adam` — L2-regularized Adam (`tensor::adam_step`)
  - `AdamW` — decoupled weight decay (`tensor::adamw_step`)
  - `fused_adam_step` / `fused_adamw_step` — one-shot helpers (like `fused_sgd_step`)
- **Tests**: `torch_nntile/tests/test_adam_step_parity.py` — single-step kernel parity, multi-step optimizer parity, and fused-step helpers (10 tests).

## Usage

```python
from torch_nntile.training import Adam, AdamW, fused_adamw_step

optimizer = AdamW(model.parameters(), lr=3e-4, weight_decay=0.1)
loss.backward()
optimizer.step()
```

## Verification

```bash
pytest -vv torch_nntile/tests/test_adam_step_parity.py
```

All 10 new tests pass against a CPU reference implementation of the libnntile Adam/AdamW update formulas.

<div><a href="https://cursor.com/agents/bc-0885c501-8e61-4fa7-9e22-fc2effe8c500"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-0885c501-8e61-4fa7-9e22-fc2effe8c500"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review

**Comment by Muxas** [comment]
@cursor review
